skyportal/handlers/api/galaxy.py

```diff
@@ -21,6 +21,126 @@
 MAX_GALAXIES = 10000
 
 
+def get_galaxies(
+    user,
+    session,
+    catalog_name=None,
+    localization_dateobs=None,
+    localization_name=None,
+    localization_cumprob=None,
+    includeGeoJSON=False,
+    catalog_names_only=False,
+    page_number=1,
+    num_per_page=MAX_GALAXIES,
+):
+    if catalog_names_only:
+        catalogs = session.execute(sa.select(Galaxy.catalog_name).distinct()).all()
+        query_result = []
+        for (catalog_name,) in catalogs:
+            query = Galaxy.query_records_accessible_by(user, mode="read")
+            query = query.filter(Galaxy.catalog_name == catalog_name)
+            total_matches = query.count()
+            query_result.append(
+                {
+                    'catalog_name': catalog_name,
+                    'catalog_count': int(total_matches),
+                }
+            )
+
+        return query_result
+
+    query = Galaxy.query_records_accessible_by(user, mode="read")
```

Comment:
Can we use Guy's session syntax here and below?

Reply:
Of course. However, I'm not sure which syntax to use exactly. Things are done differently in so many handlers that I can't tell which one is the right one to use.